gluten-core/src/main/scala/io/glutenproject/expression/TernaryExpressionTransformer.scala

```diff
@@ -108,6 +108,51 @@ case class StringLocateTransformer(
   }
 }
 
+case class StringSplitTransformerBase(
+    substraitExprName: String,
+    srcExpr: ExpressionTransformer,
+    regexExpr: ExpressionTransformer,
+    limitExpr: ExpressionTransformer,
+    original: StringSplit)
+  extends ExpressionTransformer {
+
+  override def doTransform(args: java.lang.Object): ExpressionNode = {
+    TernaryExpressionTransformer(substraitExprName, srcExpr, regexExpr, limitExpr, original)
+      .doTransform(args)
+  }
+}
+
+class StringSplitTransformer(
+    substraitExprName: String,
+    srcExpr: ExpressionTransformer,
+    regexExpr: ExpressionTransformer,
+    limitExpr: ExpressionTransformer,
+    original: StringSplit)
+  extends StringSplitTransformerBase(substraitExprName, srcExpr, regexExpr, limitExpr, original)
+  with Logging {
+
+  override def doTransform(args: java.lang.Object): ExpressionNode = {
+    val limitNode = limitExpr.doTransform(args)
+    val regexNode = regexExpr.doTransform(args)
+    if (!limitNode.isInstanceOf[LiteralNode] || !regexNode.isInstanceOf[LiteralNode]) {
+      throw new UnsupportedOperationException(
+        s"limit and regex args supported LiterNode, " +
+          s"but given ${limitNode.getClass.toString} and ${limitNode.getClass.toString}")
```

Comment:
Nit:
The fallback message should be brief and informative. I think it can be refined, e.g., "Gluten only supports literal input as limit/regex for split function." Thanks!

Reply:
Done